tuta-sdk/rust/src/user_facade.rs

```diff
@@ -1,26 +1,30 @@
-use std::sync::Arc;
+use std::sync::{Arc, RwLock};
+
 use crate::ApiCallError;
+use crate::crypto::Aes256Key;
 use crate::entities::sys::User;
-use crate::generated_id::GeneratedId;
-use crate::typed_entity_client::TypedEntityClient;
-
 
-/// FIXME: for testing unencrypted entity downloading. Remove after everything works together.
 #[derive(uniffi::Object)]
 pub struct UserFacade {
-    entity_client: Arc<TypedEntityClient>,
+    user: Arc<RwLock<User>>,
 }
 
+/// UserFacade is tied to a logged in user.
 impl UserFacade {
-    pub fn new(entity_client: Arc<TypedEntityClient>) -> Self {
-        UserFacade { entity_client }
+    pub fn new(
+        user: User,
+    ) -> Self {
+        UserFacade {
+            user: Arc::new(RwLock::new(user)),
+        }
     }
-}
 
-#[uniffi::export]
-impl UserFacade {
-    /// Gets a user (an entity/instance of `User`) from the backend
-    pub async fn load_user_by_id(&self, id: &GeneratedId) -> Result<User, ApiCallError> {
-        self.entity_client.load(id).await
+    pub fn get_user(&self) -> User {
+        self.user.read().unwrap().clone()
     }
-}
\ No newline at end of file
+
+    // FIXME
```

Comment:
shouldn't have FIXME if we are going to merge this

Reply:
It needs to be merged after `KeyLoaderFacade`. I'll rebase once the other branch is merged.